wpilibjExamples/src/main/java/edu/wpi/first/wpilibj/examples/commandbasedbestpracticeled/subsystems/RobotSignalsLEDbufferLEDSubsystem.java

```diff
@@ -0,0 +1,29 @@
+// Copyright (c) FIRST and other WPILib contributors.
+// Open Source Software; you can modify and/or share it under the terms of
+// the WPILib BSD license file in the root directory of this project.
+
+package frc.robot.subsystems;
+
+import edu.wpi.first.wpilibj.AddressableLED;
+import edu.wpi.first.wpilibj.AddressableLEDBuffer;
+import frc.robot.PeriodicTask;
+
+public class RobotSignalsLEDbufferLEDSubsystem  {
```

Comment:
Not sure why such a verbose name, also I would not include `Subsystem` suffix since this is no longer a subsystem. How about `LEDResource`?

Reply:
I apologize for leaving some trash from the start and then got distracted trying to get output from the example and getting it into a WPILib Pull Request.

My refactored version is available [here](https://github.com/tom131313/allwpilib/tree/main/wpilibjExamples/src/main/java/edu/wpi/first/wpilibj/examples/commandbasedbestpracticeled). Believe it or not it does resemble the original except I have added working output with commentary and didn't pursue the anticipated new AddressableLED functions which we can add when they are readily available. (Xbox controller buttons `A`, `B`, and `X` do something in different robot modes. Now I see that I should document the expected behavior.)

I can't remember - are we including the word `subsystem` in the Java class/file name or not? I left in `subsystem` but scrubbed `Command` and `Trigger` from names.)

I would appreciate your and @Oblarg comments on what can be done better as a best practice. Thanks.

Reply:
@illinar I have completed a working example LED resource/subsystem/view based about 99% on your example code.

I did find @SamCarlberg nice addition to the LED process that you recommended. I copied his entire work into my example since I don't see any other way to reference it in my code. All those files would have to be deleted from my example and appropriate `imports` made to the WPILib when it has those files available.

My changes to your code are trivial to make it a little easier to test. I simulated a couple of events with Xbox controller button presses and changed an `onTrue` to `whileTrue`. I added a third view of the LEDs only to display your dynamic Enable/Disable Patterns.

Since the basis of all LED addressable views is the same I have only one class for the LEDview and instantiate it three times for the Top, Main, and EnableDisable views.

I welcome comments from all - you, @Oblarg and anyone to further refine structure.

Thanks for your contribution of a fine example.